### DISCUSSION TITLE
Roslyn api for identifying a subpattern

### DISCUSSION THREAD

**Discussion by kharai-tor** [dotnet/roslyn#75512] [Q&A] [answered]
The following switch statement:

```C#
switch (x)
{
    case object o:
        break;
    case int i:
        break;
}
```
gives an error on `case int i` saying something like: `The case has already been handled by a previous case`.

Is there a publicly available method in Roslyn/Microsoft.CodeAnalysis.CSharp that, similarly to the example above, lets me know if pattern A is a subpattern of pattern B?

Can you help me with this? I'm building a very cool analyzer that you guys are all going to love!

Well, maybe not really but can you still help me please?

A huge thanks in advance!

**Comment by 333fred** [comment, marked as answer]
There is no public API to identify pattern subsumption at this time.

**Reply by kharai-tor** [reply]
Oh that's such a bummer :(

I guess the next step is to open a feature request? Can you advise me on whether it's better to do as a discussion or as an issue?

Thanks for such a quick response!

**Reply by 333fred** [reply]
Yes, you would need to propose an API shape, following the process outlined [here](https://github.com/dotnet/roslyn/blob/main/docs/contributing/API%20Review%20Process.md).
